src/Microsoft.SqlTools.ServiceLayer/Rename/RenameService.cs

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Rename.Requests;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename
+{
+    /// <summary>
+    /// Main class for Rename Service functionality
+    /// </summary>
+    public class RenameService : IDisposable
```

Comment:
@alanrenmsft I'm trying to decide here if we should make a more generic service that is used for generic modifications to objects, instead of a specialized service just for a single function. Thoughts on that? 

(and ideally making it something that other providers could implement as well)

Reply:
I thought about it, we probably should call it `ObjectManagementProvider` in dataprotocol  and in  sqltoolsservice, we can call the service `ObjectManagemetnService`), it is ok to start with only the rename functionality.  other features like create, retrieval, update and delete will be added later, this is something we are planning to do in the new semester.

Reply:
Yeah, something like that long term sounds great for sure. I figure we can get this logic in as just a single-purpose service for just MSSQL for now and then later on when/if we make it generic in the data protocol or whatever we can move the logic over there and fully implement that.

Reply:
@Charles-Gagnon and @alanrenmsft is it ok, if I rename the ```RenameService``` to ```ObjectManagementService```?

Reply:
Yeah, that sounds great to me. And then the request ID would be `objectmanagement/rename` as Alan suggested